src/main/java/com/google/sps/utils/GameUtils.java

```diff
@@ -123,4 +124,59 @@ public static boolean addUserToGame(String userId, Entity gameEntity, DatastoreS
     
     return true;
   }
-}
\ No newline at end of file
+ 
+  public static String joinGame(Entity userEntity, String gameName, DatastoreService datastore, String gameId) {
+ 
+    if(gameName == ""){
+        return "noGameName";
+    }
+ 
+    if(userEntity == null){
+        log.severe("found null userEntity trying to join game " + gameName);
+        return "nullUserEntity";
+    }
+ 
+    if(datastore == null){
+        log.severe("found null datastore trying to join game " + gameName);
+        return "nullDatastore";
+    }
+ 
+    // prevent users from joining more than one game
+    if(userEntity.getProperty("gameId") != null){
+      return "alreadyHasGame";
+    }
+ 
+    Entity gameEntity;
+    if(gameId == null){
+        // create game entity
+        gameEntity = GameUtils.createGameEntity(gameName, datastore);
+        if(gameEntity == null){
+            log.severe("create game failed for game " + gameName);
+            return "createGameFailed";
+        }
+ 
+    }else{
+        // or join a game 
+        gameEntity = UserUtils.getEntityFromDatastore("Game", "gameId", gameId, datastore);
+    
+        if(gameEntity == null){ 
+          return "badGameCode";
+        }
+    }
+ 
+    // add user to game entity + vice versa
+    boolean userAdded = GameUtils.addUserToGame((String) userEntity.getProperty("userId"), gameEntity, datastore);
+    if(!userAdded){
+        log.severe("failed to add user to game " + gameName);
```

Comment:
Which user? could be useful to log the user ID

Reply:
I added the user to these.